lib/ansible/modules/packaging/os/dnf.py

```diff
@@ -946,14 +946,7 @@ def ensure(self):
                             self.module_base.install([module])
                             self.module_base.enable([module])
                         except dnf.exceptions.MarkingErrors as e:
-                            failure_response['failures'].append(
-                                " ".join(
-                                    (
-                                        ' '.join(module),
-                                        to_native(e)
-                                    )
-                                )
-                            )
+                            failure_response['failures'].append(' '.join(module, to_native(e)))
```

Comment:
`join` takes one argument, so pass a tuple:
```suggestion
                            failure_response['failures'].append(" ".join((module, to_native(e))))
```

Same below.